app/main.py

```diff
@@ -1,2 +1,113 @@
+from __future__ import annotations
+from typing import Any, Hashable, Iterator
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    hash_: int
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.__size = 0
+        self.__capacity = 8
+        self._hash_table: list[Node | None] = [None] * self.__capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        key_hash = hash(key)
+        hash_index = self.__index_to_write(key)
+        if self._hash_table[hash_index] is None:
+            self.__size += 1
+        node = Node(hash_=key_hash, key=key, value=value)
+        self._hash_table[hash_index] = node
+
+        if len(self) > int(self.__capacity * (2 / 3)):
+            self.__resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_index = self.__index_to_read(key)
+        return self._hash_table[hash_index].value
+
+    def __len__(self) -> int:
+        return self.__size
+
+    def __delitem__(self, key: Hashable) -> None:
+        hash_index = self.__index_to_read(key)
+        self._hash_table[hash_index] = None
+        self.__size -= 1
+
+    def __repr__(self) -> str:
+        str_ = "{"
+        for el in self:
+            str_ += f"'{el}': {self.get(el)}, "
+        return str_[:-2] + "}"
```

Comment:
The `__repr__` method may raise an error if the dictionary is empty because it attempts to slice the string `str_[:-2]`. Consider checking if the dictionary is empty before slicing to avoid this issue.